lib/libzfs_core/libzfs_core.c

```diff
@@ -645,6 +656,90 @@ lzc_send_resume(const char *snapname, const char *from, int fd,
 	    resumeoff, NULL));
 }
 
+unsigned long
+lzc_get_pipe_max()
+{
+	unsigned long max_psize = 65536;
+#ifdef __linux__
+	FILE *procf = fopen("/proc/sys/fs/pipe-max-size", "re");
+
+	if (procf != NULL) {
+		if (fscanf(procf, "%lu", &max_psize) <= 0) {
+			// safe default
+			max_psize = 65536;
+		}
+		fclose(procf);
+	}
+#endif
+	return (max_psize);
+}
+
+void
+lzc_set_pipe_max(int infd)
+{
+#ifdef __linux__
```

Comment:
something like the "FreeBSD automatically resizes" comment from the libzfs impl would be good here as well

Reply:
actually, a comment along the lones of "FreeBSD grows automatically to 64k" on the `unsigned long max_psize = 65536;` line would kill both birds in one chuck maybe?

Reply:
Sure, done.